File: project/silver-layer/src/main/java/csx55/sta/silver/SilverLayerApp.java
package csx55.sta.silver;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Main entry point for Silver Layer transformations
 *
 * Silver layer performs:
 * - Data cleaning and validation
 * - Data enrichment and joins
 * - Deduplication
 * - Schema evolution handling
 * - Business logic transformations
 *
 * Example jobs to implement:
 * - CleanedTripsJob: Clean and validate trip data
 * - EnrichedTripsJob: Join trips with weather and events
 * - DedupTripsJob: Remove duplicate trip records
 */
public class SilverLayerApp {
    private static final Logger logger = LoggerFactory.getLogger(SilverLayerApp.class);

    public static void main(String[] args) {
        logger.info("Silver Layer - Not yet implemented");
        logger.info("Future jobs:");
        logger.info("  - CleanedTripsJob: Validate and clean trip data");
        logger.info("  - EnrichedTripsJob: Join trips + weather + events");
        logger.info("  - DedupTripsJob: Deduplicate records");

        System.out.println("\n=== Silver Layer Jobs (To Be Implemented) ===\n");
        System.out.println("Suggested implementations:");
        System.out.println("  1. Read from lakehouse.bronze.* tables");
        System.out.println("  2. Apply data quality rules and validations");
        System.out.println("  3. Join trips with weather and events by location + time");
        System.out.println("  4. Write to lakehouse.silver.* tables");
        System.out.println("\nExtend BaseStreamingJob from streaming-common module.");
    }
}

File: project/lakehouse/silver/src/main/scala/csx55/sta/silver/SilverLayerApp.scala
package csx55.sta.silver

import csx55.sta.silver.jobs.{EventsCleanedJob, TripsCleanedJob, WeatherCleanedJob}
import csx55.sta.streaming.config.StreamConfig
import org.slf4j.LoggerFactory

object SilverLayerApp {
  private val logger = LoggerFactory.getLogger(this.getClass)

  def main(args: Array[String]): Unit = {
    if (args.length == 0) {
      logger.error("No job name provided.")
      System.exit(1)
    }

    val jobName = args(0)
    val config = new StreamConfig()

    jobName.toLowerCase match {
      case "trips-cleaned" =>
        new TripsCleanedJob(config).run()

      case "weather-cleaned" =>
        new WeatherCleanedJob(config).run()

      case "events-cleaned" =>
        new EventsCleanedJob(config).run()

      case _ =>
        logger.error("Unknown job: {}.", jobName)
        System.exit(1)
    }
  }
}
